harness/determined/pytorch/lightning/_adapter.py

```diff
@@ -152,16 +171,37 @@ def __init__(self, context: PyTorchTrialContext) -> None:
         optimizers, _ = self.setup_optimizers_schedulers()
         pls.optimizers = optimizers
 
+        # CHECK our `check` usage vs raising exceptions seems inconsistent.
+        check.check_in(precision, {16, 32}, "only precisions 16 & 32 are supported.")
+
+        use_amp = context.experimental._auto_amp or context._use_apex
+
         # set lightning_module properties
         pls.lm.use_ddp = False
         pls.lm.use_ddp2 = False
         pls.lm.use_dp = False
         pls.lm.use_tpu = False
         type(pls.lm).local_rank = context.distributed.get_local_rank()  # type: ignore
         type(pls.lm).global_rank = context.distributed.get_rank()  # type: ignore
-        pls.lm.use_amp = context.experimental._auto_amp or context._use_apex
+        pls.lm.use_amp = use_amp
+        pls.lm.precision = "mixed" if use_amp else precision  # type: ignore
```

Comment:
Does this have the same behavior as `LightningModule.precision`?

Reply:
yeah if I think so. through `accelerator.precision`. I'll share the reference

Reply:
AFAICT comes from `class MixedPrecisionPlugin(PrecisionPlugin):` base class